### ISSUE TITLE
Allow selection of ends of Revolve and Helix

### ISSUE THREAD

**Issue by phkahler** [solvespace/solvespace#467]
The ends of extrusions are flat surfaces that can be selected. Points can be constrained to those surfaces. The same should be possible for objects created with Revolve and Helix.

I tried adding the following to SShell::MakeFromHelicalRevolutionOf

`

        hEntity face = Entity::NO_ENTITY;

        hEntity hface = group->Remap(face, Group::REMAP_BOTTOM);

        s0.face = hface.v;

`
Which made it crash when hovering over one end.

**Comment by phkahler** [comment]
@whitequark I made a branch "end_surfaces" in my repo that adds some Remap magic. I copied the function Group::MakeExtrusionTopBottomFaces to create one Group::MakeRevolveEndFaces. This does the same Remap stuff. I also added the above Remap stuff to the code that creates the extrusion - rather than finding those two surfaces again later like extrude does. Anyway, this allows a mouse hover to highlight the end surfaces. But it crashes if you click on them.

I wandered back into this issue while working on something else in Entity.cpp and looking at surface entities. For one, I don't see where and surface entities are actually created extrusion end caps. So maybe I didn't copy enough, but I don't know where else to look.

Also, any good writeup on how Remap actually works would be nice. Maybe an old thread has a good description?
